File: frontend/src/modules/auth/services/permissionService.ts
import { apiGet } from '@/services/apiClient';
import type { PermissionGroupResponse, PermissionResponse } from '../types';

/** Backend: auth/controller/PermissionController.java */
export const permissionService = {
  list: () => apiGet<PermissionResponse[]>('/v1/permissions'),

  grouped: () => apiGet<PermissionGroupResponse[]>('/v1/permissions/grouped'),
};

File: backend/src/main/java/com/hardware/erp/auth/service/PermissionService.java
package com.hardware.erp.auth.service;

import com.hardware.erp.auth.dto.PermissionGroupResponse;
import com.hardware.erp.auth.dto.PermissionResponse;

import java.util.List;

public interface PermissionService {

    List<PermissionResponse> findAll();

    List<PermissionGroupResponse> findAllGroupedByModule();
}
